src/hawkmoth/parser.py

```diff
@@ -185,6 +186,27 @@ def _comment_extract(tu):
 
     return top_level_comments, comments
 
+# Workaround for clang cursor.get_tokens() being unreliable for cursors whose
+# extent contains macro expansions. The result may be empty or contain bogus
+# tokens, depending on the case.
+#
+# The problem seems to be related to cursor.extent. Recreating the extent and
+# getting the tokens from the translation unit works fine. The __repr__ for both
+# the recreated and original extents is the same, but comparison indicates they
+# do differ under the hood.
```

Comment:
We've used this pattern before, but increasingly I think we should place these kind of comments as docstrings of the symbol itself, because why not? Anyway, not really an issue so it's up to you.

Reply:
I think I'll skip this for for now, but overall I agree. And really, the only way for the docstrings to be useful is to actually incorporate them into developer documentation. Not sure if it needs to be part of the regular `make html` build, or maybe hidden behind some knob.

Reply:
Well, not necessarily... If you were doing something with this on a console as a dev, you could `help('symbol_name')` and such. But it's totally fine.

Reply:
That's just a testament to the fact that I never use `help` like that... and I probably should!